extensions/amp-form/0.1/amp-form.js

```diff
@@ -274,8 +274,13 @@ export class AmpForm {
   renderTemplate_(data = {}) {
     const container = this.form_.querySelector(`[${this.state_}]`);
     if (container) {
+      const randId = `rendered-message-${Math.random() * 1000}`;
```

Comment:
This is gonna create hard-to-reproduce bugs on collision. Let's create a simple Id service with an internal singleton counter. Maybe `getIdGeneratorFor(this.win).getId();  //returns 'amp-id-<n>'`